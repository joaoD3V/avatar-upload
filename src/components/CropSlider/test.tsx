import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import { CropSlider } from '.';

describe('<AvatarEditing />', () => {
  it('should render title and slider correctly', () => {
    const handleUpdateSliderValue = jest.fn();
    const { container } = renderWithTheme(
      <CropSlider
        handleUpdateSliderValue={handleUpdateSliderValue}
        sliderValue={0}
      />
    );

    expect(screen.getByText(/Crop/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Crop slider/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
