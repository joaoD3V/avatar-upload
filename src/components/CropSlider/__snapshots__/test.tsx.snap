// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CropSlider /> should render title and slider correctly 1`] = `
.c0 {
  width: 100%;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start !important;
  -webkit-box-align: flex-start !important;
  -ms-flex-align: flex-start !important;
  align-items: flex-start !important;
  -webkit-box-pack: start !important;
  -webkit-justify-content: flex-start !important;
  -ms-flex-pack: start !important;
  justify-content: flex-start !important;
}

.c1 {
  font-size: 400;
  font-size: 1.6rem;
  color: #677489;
}

.c2 {
  width: 100%;
  padding: 0 4px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  margin-top: 14px;
  position: relative !important;
}

.c2 .rc-slider {
  position: relative;
  width: 100%;
  height: 14px;
  padding: 5px 0;
  border-radius: 6px;
  touch-action: none;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c2 .rc-slider * {
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c2 .rc-slider-rail {
  position: absolute;
  width: 100%;
  height: 2px;
  background-color: #a5c3ff;
  border-radius: 6px;
  pointer-events: none !important;
}

.c2 .rc-slider-track {
  position: absolute;
  height: 2px;
  background-color: #3F80FF;
  border-radius: 6px;
}

.c2 .rc-slider-handle {
  position: absolute;
  width: 12px;
  height: 12px;
  margin-top: -5px;
  background-color: #3F80FF;
  border-radius: 50%;
  cursor: pointer;
  cursor: -webkit--webkit-grab;
  cursor: -moz--webkit-grab;
  cursor: -webkit-grab;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  touch-action: pan-x;
}

.c2 .rc-slider-handle-dragging.rc-slider-handle-dragging.rc-slider-handle-dragging {
  border-color: #57c5f7;
}

.c2 .rc-slider-handle:focus {
  outline: none;
  box-shadow: none;
}

.c2 .rc-slider-handle:focus-visible {
  border-color: #2db7f5;
}

.c2 .rc-slider-handle-click-focused:focus {
  border-color: #96dbfa;
  box-shadow: unset;
}

.c2 .rc-slider-handle:hover {
  border-color: #57c5f7;
}

.c2 .rc-slider-handle:active {
  border-color: #57c5f7;
  cursor: -webkit--webkit-grabbing;
  cursor: -moz--webkit-grabbing;
  cursor: -webkit-grabbing;
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c2 .rc-slider-mark {
  position: absolute;
  top: 18px;
  left: 0;
  width: 100%;
  font-size: 12px;
}

.c2 .rc-slider-mark-text {
  position: absolute;
  display: inline-block;
  color: #999;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
}

.c2 .rc-slider-mark-text-active {
  color: #666;
}

.c2 .rc-slider-step {
  position: absolute;
  width: 100%;
  height: 2px;
  background: transparent;
  pointer-events: none;
}

.c2 .rc-slider-dot {
  position: absolute;
  bottom: -2px;
  width: 8px;
  height: 8px;
  vertical-align: middle;
  background-color: #fff;
  border: 2px solid #e9e9e9;
  border-radius: 50%;
  cursor: pointer;
}

.c2 .rc-slider-dot-active {
  border-color: #96dbfa;
}

.c2 .rc-slider-dot-reverse {
  margin-right: -4px;
}

.c2 .rc-slider-disabled {
  background-color: #e9e9e9;
}

.c2 .rc-slider-disabled .rc-slider-track {
  background-color: #ccc;
}

.c2 .rc-slider-disabled .rc-slider-handle,
.c2 .rc-slider-disabled .rc-slider-dot {
  background-color: #fff;
  border-color: #ccc;
  box-shadow: none;
  cursor: not-allowed;
}

.c2 .rc-slider-disabled .rc-slider-mark-text,
.c2 .rc-slider-disabled .rc-slider-dot {
  cursor: not-allowed !important;
}

.c2 .rc-slider-vertical {
  width: 14px;
  height: 100%;
  padding: 0 5px;
}

.c2 .rc-slider-vertical .rc-slider-rail {
  width: 4px;
  height: 100%;
}

.c2 .rc-slider-vertical .rc-slider-track {
  bottom: 0;
  left: 5px;
  width: 4px;
}

.c2 .rc-slider-vertical .rc-slider-handle {
  margin-top: 0;
  margin-left: -5px;
  touch-action: pan-y;
}

.c2 .rc-slider-vertical .rc-slider-mark {
  top: 0;
  left: 18px;
  height: 100%;
}

.c2 .rc-slider-vertical .rc-slider-step {
  width: 4px;
  height: 100%;
}

.c2 .rc-slider-vertical .rc-slider-dot {
  margin-left: -2px;
}

.c2 .rc-slider-tooltip-zoom-down-enter,
.c2 .rc-slider-tooltip-zoom-down-appear {
  display: block !important;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
}

.c2 .rc-slider-tooltip-zoom-down-leave {
  display: block !important;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
}

.c2 .rc-slider-tooltip-zoom-down-enter.rc-slider-tooltip-zoom-down-enter-active,
.c2 .rc-slider-tooltip-zoom-down-appear.rc-slider-tooltip-zoom-down-appear-active {
  -webkit-animation-name: rcSliderTooltipZoomDownIn;
  animation-name: rcSliderTooltipZoomDownIn;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2 .rc-slider-tooltip-zoom-down-leave.rc-slider-tooltip-zoom-down-leave-active {
  -webkit-animation-name: rcSliderTooltipZoomDownOut;
  animation-name: rcSliderTooltipZoomDownOut;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2 .rc-slider-tooltip-zoom-down-enter,
.c2 .rc-slider-tooltip-zoom-down-appear {
  -webkit-transform: scale(0,0);
  -ms-transform: scale(0,0);
  transform: scale(0,0);
  -webkit-animation-timing-function: cubic-bezier(0.23,1,0.32,1);
  animation-timing-function: cubic-bezier(0.23,1,0.32,1);
}

.c2 .rc-slider-tooltip-zoom-down-leave {
  -webkit-animation-timing-function: cubic-bezier(0.755,0.05,0.855,0.06);
  animation-timing-function: cubic-bezier(0.755,0.05,0.855,0.06);
}

.c2 .rc-slider-tooltip {
  position: absolute;
  top: -9999px;
  left: -9999px;
  visibility: visible;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c2 .rc-slider-tooltip * {
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.c2 .rc-slider-tooltip-hidden {
  display: none;
}

.c2 .rc-slider-tooltip-placement-top {
  padding: 4px 0 8px 0;
}

.c2 .rc-slider-tooltip-inner {
  min-width: 24px;
  height: 24px;
  padding: 6px 2px;
  color: #fff;
  font-size: 12px;
  line-height: 1;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #6c6c6c;
  border-radius: 6px;
  box-shadow: 0 0 4px #d9d9d9;
}

.c2 .rc-slider-tooltip-arrow {
  position: absolute;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
}

.c2 .rc-slider-tooltip-placement-top .rc-slider-tooltip-arrow {
  bottom: 4px;
  left: 50%;
  margin-left: -4px;
  border-width: 4px 4px 0;
  border-top-color: #6c6c6c;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Crop
  </span>
  <div
    class="c2"
  >
    <div
      class="rc-slider rc-slider-horizontal"
    >
      <div
        class="rc-slider-rail"
      />
      <div
        class="rc-slider-track"
        style="left: 0%; width: 0%;"
      />
      <div
        class="rc-slider-step"
      />
      <div
        aria-disabled="false"
        aria-label="Crop slider"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle"
        role="slider"
        style="left: 0%; transform: translateX(-50%);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
